src/app/preferences/general-preferences/general-preferences.service.ts

```diff
@@ -0,0 +1,18 @@
+namespace Core {
+  export class GeneralPreferencesService {
+    constructor(private $window: ng.IWindowService) {
+      'ngInject';
+      this.$window.localStorage.setItem('defaultVerticalNavState', 'show');
```

Comment:
Hmm, I think I found a bug here. With this code in the constructor, doesn't the class always overwrite what the user configured before to `show`?  I suspect it means any time you reload the page it always start with the `show` state, doesn't it?

Reply:
Yes it does. The default selected option is show. I tried running the code without the constructor, show was not woking. Let me check it once more.

Reply:
ok, you can look at how other preferences implement this
https://github.com/hawtio/hawtio-core/blob/main/src/app/preferences/logging-preferences/logging-preferences.service.ts

Reply:
Okay makes sense. I'll try implementing like this

Reply:
so it doesn't set anything by default at constructor. And the getter method returns something other than the default only when it's set in local storage. And don't forget to define `DEFAULT_VERTICAL_NAV_STATE` variable as a static readonly constant.
https://www.typescriptlang.org/docs/handbook/2/classes.html#readonly